src/Microsoft.Framework.Primitives/StringValues.cs

```diff
@@ -4,13 +4,14 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using Microsoft.Framework.Internal;
 
 namespace Microsoft.Framework.Primitives
 {
     /// <summary>
     /// Represents zero/null, one, or many strings in an efficient way.
     /// </summary>
-    public struct StringValues : IList<string>
+    public struct StringValues : IList<string>, IReadOnlyList<string>
```

Comment:
No, we want to maintain interop with `IList<string>` because that's what string[] implements. It avoids more conversions. Many more APIs take IList than IReadOnlyList, including most of Linq.